lib/ansible/plugins/connection/persistent.py

```diff
@@ -98,11 +99,20 @@ def _start_connection(self):
             raise AnsibleError("Unable to find location of 'ansible-connection'. "
                                "Please set or check the value of ANSIBLE_CONNECTION_PATH")
 
+        env = os.environ.copy()
+        env.update({
+                   'ANSIBLE_BECOME_PLUGINS': become_loader.print_paths(),
+                   'ANSIBLE_CLICONF_PLUGINS': cliconf_loader.print_paths(),
+                   'ANSIBLE_CONNECTION_PATH': connection_loader.print_paths(),
```

Comment:
Shouldn't this be `ANSIBLE_CONNECTION_PLUGINS`? `ANSIBLE_CONNECTION_PATH` is the path to `ansible-connection` itself, not the envvar used by the loader.

Reply:
at least one of us had coffee today ...